modules/seqexec/server/src/main/scala/seqexec/server/gmos/GmosInstrumentActions.scala

```diff
@@ -135,22 +140,31 @@ class GmosInstrumentActions[F[_]: MonadError[?[_], Throwable]: Logger, A <: Gmos
    * Calculate the actions for a N&S cycle
    */
   private def actionPositions(
-    env:       ObserveEnvironment[F],
-    rows:      Int,
-    post:      (Stream[F, Result[F]], ObserveEnvironment[F]) => Stream[F, Result[F]]
+    env:   ObserveEnvironment[F],
+    rows:  Int,
+    positions: Vector[NSPosition],
+    post:  (Stream[F, Result[F]], ObserveEnvironment[F]) => Stream[F, Result[F]]
   ): List[ParallelActions[F]] =
     Gmos.NsSequence.zipWithIndex
       .map {
         case (stage, i) =>
           val rowsToShuffle = if (stage === StageA) 0 else rows
+          val nsPositionO = positions.find(_.stage === stage)
           List(
             // Configure rows to shuffle
-            NonEmptyList.one(
+            NonEmptyList(
               inst
                 .configureShuffle(rowsToShuffle)
                 .as(Response.Configured(inst.resource))
-                .toAction(ActionType.Configure(inst.resource))
-              // TODO confgure the TCS
+                .toAction(ActionType.Configure(inst.resource)),
+              (tcsO, nsPositionO).mapN{ case (tcs, nsPos) =>
```

Comment:
How is it in case of a dark? Would `nsPosition0` be `None`?

Reply:
I think we need a new type of step, but the implementation will be easy.